Refresh quickstart guide on Rootstock (#967)
rsksmart/devportal: content/rsk-devportal/guides/quickstart/hardhat/test-smart-contract.md

---
menu_order: 400
menu_title: Test Smart Contracts
layout: rsk
title: 'Test Smart Contracts'
description: 'Learn how to test your Rootstock smart contract'
tags: quick-start, getting-started, guide, how-to, bitcoin, rsk, rootstock, blockchain
---

In this section, you'll set up a smart contract test and test your contract to ensure it works as expected.

> Follow these stepas to test your smart contract.

Reviewer: ```suggestion
Follow these steps below to test your smart contract.
```

### Step 1: Install Dependencies

We'll install the Mocha and Chai testing dependencies.

Mocha is a JavaScript test framework running on Node.js. Chai is an assertion library for the browser and Node that can be paired with any JavaScript testing framework.

- Before writing tests for your token contract, install the required testing dependencies:
  ```shell
  npm install --save-dev mocha@10.2.0 chai@4.2.0 @nomiclabs/hardhat-ethers@2.2.3
  ```

### Step 2: Create Tests

> 1. Navigate to the `test` directory in the root directory of your project, this is where we recommend storing all test files:

Reviewer: ```suggestion
1. Navigate to the `test` directory in the root directory of your project, this is recommended for storing all test files:
```
